src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.cs

```diff
@@ -1311,8 +1311,183 @@ public static Matrix4x4 CreateReflection(Plane value)
         /// <param name="matrix">The source matrix to invert.</param>
         /// <param name="result">If successful, contains the inverted matrix.</param>
         /// <returns>True if the source matrix could be inverted; False otherwise.</returns>
-        public static bool Invert(Matrix4x4 matrix, out Matrix4x4 result)
+        public static unsafe bool Invert(Matrix4x4 matrix, out Matrix4x4 result)
         {
+            if (Sse.IsSupported)
+            {
+                // This implementation is based on the DirectX Math Library XMMInverse method
+                // https://github.com/microsoft/DirectXMath/blob/master/Inc/DirectXMathMatrix.inl
+
+                // Load the matrix values into rows
+                Vector128<float> row1 = Sse.LoadVector128(&matrix.M11);
+                Vector128<float> row2 = Sse.LoadVector128(&matrix.M21);
+                Vector128<float> row3 = Sse.LoadVector128(&matrix.M31);
+                Vector128<float> row4 = Sse.LoadVector128(&matrix.M41);
+
+                // Transpose the matrix
+                Vector128<float> vTemp1 = Sse.Shuffle(row1, row2, 0x44);
+                Vector128<float> vTemp3 = Sse.Shuffle(row1, row2, 0xEE);
+                Vector128<float> vTemp2 = Sse.Shuffle(row3, row4, 0x44);
+                Vector128<float> vTemp4 = Sse.Shuffle(row3, row4, 0xEE);
+
+                row1 = Sse.Shuffle(vTemp1, vTemp2, 0x88);
+                row2 = Sse.Shuffle(vTemp1, vTemp2, 0xDD);
+                row3 = Sse.Shuffle(vTemp3, vTemp4, 0x88);
+                row4 = Sse.Shuffle(vTemp3, vTemp4, 0xDD);
+
+                Vector128<float> V00 = Sse.Shuffle(row3, row3, 0x50);
+                Vector128<float> V10 = Sse.Shuffle(row4, row4, 0xEE);
+                Vector128<float> V01 = Sse.Shuffle(row1, row1, 0x50);
+                Vector128<float> V11 = Sse.Shuffle(row2, row2, 0xEE);
+                Vector128<float> V02 = Sse.Shuffle(row3, row1, 0x88);
+                Vector128<float> V12 = Sse.Shuffle(row4, row2, 0xDD);
+
+                Vector128<float> D0 = Sse.Multiply(V00, V10);
+                Vector128<float> D1 = Sse.Multiply(V01, V11);
+                Vector128<float> D2 = Sse.Multiply(V02, V12);
+
+                V00 = Sse.Shuffle(row3, row3, 0xEE);
+                V10 = Sse.Shuffle(row4, row4, 0x50);
+                V01 = Sse.Shuffle(row1, row1, 0xEE);
+                V11 = Sse.Shuffle(row2, row2, 0x50);
+                V02 = Sse.Shuffle(row3, row1, 0xDD);
+                V12 = Sse.Shuffle(row4, row2, 0x88);
+
+                D0 = Sse.Subtract(D0, Sse.Multiply(V00, V10));
```

Comment:
Might be worth a comment that we deviate here and only use `Subtract(op, Multiply(op, op))`. It probably needs further discussion/consideration if we can have the path deviate on hardware with FMA support as it will result in different answers in some cases.

Reply:
Added explanation to why this expansion was used.